src/components/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,55 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { Modal } from "./Modal";
+
+describe("Modal component", () => {
```

Comment:
You may also add a test to ensure focus moves to the modal when opened, test if the focus returns to the triggering element when the modal closes, include a snapshot test to ensure the modal's UI does not unintentionally change, test the scroll lock behavior, write a test for event propagation and add a test to check if the modal doesn't close when user clicks any other key than `Escape`.